intake/catalog/base.py

```diff
@@ -188,11 +206,86 @@ def _close(self):
         # TODO: maybe close all entries?
         pass
 
+class Entries(dict):
+    """Fetches entries from server on item lookup and iteration.
+
+    This fetches pages of entries from the server during iteration and
+    caches them. On __getitem__ it fetches the sepcific entry from the
+    server.
+    """
+    # This has PY3-style lazy methods (keys, values, items). Since it's
+    # internal we should not need the PY2-only iter* variants.
+    def __init__(self, catalog):
+        self._catalog = catalog
+        self._page_cache = collections.OrderedDict()
+        # Put lookups that were due to __getitem__ in a separate cache
+        # so that iteration reflects the server's order, not an
+        # arbitrary cache order.
+        self._direct_lookup_cache = {}
+        self._page_offset = 0
+
+    def reset(self):
+        "Clear caches to force a reload."
+        self._page_cache.clear()
+        self._direct_lookup_cache.clear()
+        self._page_offset = 0
+
+    def __iter__(self):
+        for key in self.keys():
+            yield key
+
+    def __contains__(self, key):
```

Comment:
Does this mean that the `__contains__` for the Catalog class can just call `if in` on its entries dict object?